pkg/cmd/secvulnReport.go

```diff
@@ -37,4 +46,208 @@ func init() {
+
+		// First structure
+		// ### CVE
+		// #### Galasa projects
+		// ##### Dependency chain
+		consolidateSecVulnYamlReports(yamlReport, "cve")
+
+		// Second structure
+		// ### Galasa project
+		// #### CVEs
+		// ##### Dependency chain
+		consolidateSecVulnYamlReports(yamlReport, "project")
+
+	}
+
+	// Write the Markdown page
+	markdownPage := "# Galasa Security Vulnerability report\n\n"
+	markdownPage += writeMarkdown()
+
+	// Write the Markdown to a file and export
+	exportMarkdownPage(markdownPage)
+}
+
+func unmarshalSecVulnYamlReports(directory string) (YamlReport, error) {
+	var yamlReport YamlReport
+
+	yamlFile, err := os.ReadFile(fmt.Sprintf("%s/%s", directory, "galasa-secvuln-report.yaml"))
+	if err != nil {
+		return yamlReport, err
+	}
+
+	err = yaml.Unmarshal(yamlFile, &yamlReport)
+	if err != nil {
+		return yamlReport, err
+	}
+
+	return yamlReport, err
+}
+
+func getAcceptanceYamlReport() (AcceptanceYamlReport, error) {
+
+	var acceptanceReport AcceptanceYamlReport
+
+	url := fmt.Sprintf("%s/%s", secvulnReportAcceptance, "override.yaml")
+	response, err := http.Get(url)
+	if err != nil {
+		return acceptanceReport, err
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return acceptanceReport, err
+	}
+
+	err = yaml.Unmarshal(body, &acceptanceReport)
+	if err != nil {
+		return acceptanceReport, err
+	}
+
+	return acceptanceReport, err
+}
+
+func consolidateSecVulnYamlReports(yamlReport YamlReport, topLevelObject string) {
+	for _, vulnerability := range yamlReport.Vulnerabilities {
+		for _, project := range vulnerability.Projects {
+
+			cve := vulnerability.Cve
+			projectName := project.Project
+			depChain := project.DependencyChain
+
+			if topLevelObject == "cve" {
+				writeToMap(firstMap, cve, projectName, depChain)
+			} else if topLevelObject == "project" {
+				writeToMap(secondMap, projectName, cve, depChain)
+			}
+		}
+	}
+}
+
+func writeToMap(whicheverMap map[string]map[string][]string, topLevelObject, secondLevelObject, depChain string) {
+	if whicheverMap[topLevelObject] != nil {
+		if whicheverMap[topLevelObject][secondLevelObject] != nil {
+			whicheverMap[topLevelObject][secondLevelObject] = append(whicheverMap[topLevelObject][secondLevelObject], depChain)
+		} else {
+			var depChainArray []string
+			depChainArray = append(depChainArray, depChain)
+			whicheverMap[topLevelObject][secondLevelObject] = depChainArray
+		}
+	} else {
+		var depChainArray []string
+		depChainArray = append(depChainArray, depChain)
+
+		projectMap := make(map[string][]string)
+		projectMap[secondLevelObject] = depChainArray
+
+		whicheverMap[topLevelObject] = projectMap
+	}
+}
+
+func getAcceptanceData(cve string) (string, string) {
+	for _, value := range acceptanceReport.Cves {
+		if value.Cve == cve {
+			return value.Comment, value.ReviewDate
+		}
+	}
+	return "", ""
+}
+
+func writeMarkdown() string {
```

Comment:
Use templating instead of "manually" creating the markdown file using string manipulation,    will make it a billion times easier to pretify the page.
https://pkg.go.dev/text/template